c7n/resources/dms.py

```diff
@@ -136,6 +136,69 @@ def process(self, resources):
             client.delete_replication_instance(ReplicationInstanceArn=arn)
 
 
+@ReplicationInstance.action_registry.register('modify-instance')
+class ModifyReplicationInstance(BaseAction):
+    """Modify replication instance(s) to apply new settings
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: enable-minor-version-upgrade
+            resource: dms-instance
+            filters:
+              - AutoMinorVersionUpgrade: False
+            actions:
+              - type: modify-instance
+                ApplyImmediately: True
+                AutoMinorVersionUpgrade: True
+                PreferredMaintenanceWindow: mon:23:00-mon:23:59
+
+    AWS ModifyReplicationInstance Documentation: https://goo.gl/ePye9N
+    """
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['modify-instance']},
+            'ReplicationInstanceArn': {'type': 'string'},
+            'AllocatedStorage': {'type': 'integer'},
+            'ApplyImmediately': {'type': 'boolean'},
+            'ReplicationInstanceClass': {'type': 'string'},
+            'VpcSecurityGroupIds': {
+                'type': 'array', 'items': {'type': 'string'}
+            },
+            'PreferredMaintenanceWindow': {'type': 'string'},
+            'MultiAZ': {'type': 'boolean'},
+            'EngineVersion': {'type': 'string'},
+            'AllowMajorVersionUpgrade': {'type': 'boolean'},
+            'AutoMinorVersionUpgrade': {'type': 'boolean'},
+            'ReplicationInstanceIdentifier': {'type': 'string'}
+        }
+    }
+    permissions = ('dms:ModifyReplicationInstance',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('dms')
+        params = dict(self.data)
+        params.pop('type')
+        for r in resources:
+            params['ReplicationInstanceArn'] = r['ReplicationInstanceArn']
+            try:
+                client.modify_replication_instance(**params)
+            except ClientError as e:
+                if e.response['Error']['Code'] in (
+                        'InsufficientResourceCapacityFault',
+                        'InvalidResourceStateFault',
+                        'ResourceAlreadyExistsFault',
+                        'ResourceNotFoundFault',
+                        'StorageQuotaExceededFault',
```

Comment:
quota exceeded is a persistent fault and should not be caught but raised, ditto for insufficientresourcecapacity,

upgradedependencyfailurefault should log.

Reply:
Ok, thanks for clarifying on those. I wasn't sure.